homeassistant/components/flipr/sensor.py

```diff
@@ -1,90 +1,89 @@
 from homeassistant.const import (
     ATTR_ATTRIBUTION,
     DEVICE_CLASS_TEMPERATURE,
     DEVICE_CLASS_TIMESTAMP,
     ELECTRIC_POTENTIAL_MILLIVOLT,
     TEMP_CELSIUS,
 )
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
 from . import FliprEntity
-from .const import ATTRIBUTION, CONF_FLIPR_ID, DOMAIN
+from .const import ATTRIBUTION, DOMAIN
 
-SENSORS = {
-    "chlorine": {
-        "unit": ELECTRIC_POTENTIAL_MILLIVOLT,
-        "icon": "mdi:pool",
-        "name": "Chlorine",
-        "device_class": None,
-    },
-    "ph": {"unit": None, "icon": "mdi:pool", "name": "pH", "device_class": None},
-    "temperature": {
-        "unit": TEMP_CELSIUS,
-        "icon": None,
-        "name": "Water Temp",
-        "device_class": DEVICE_CLASS_TEMPERATURE,
-    },
-    "date_time": {
-        "unit": None,
-        "icon": None,
-        "name": "Last Measured",
-        "device_class": DEVICE_CLASS_TIMESTAMP,
-    },
-    "red_ox": {
-        "unit": ELECTRIC_POTENTIAL_MILLIVOLT,
-        "icon": "mdi:pool",
-        "name": "Red OX",
-        "device_class": None,
-    },
-}
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="chlorine",
+        name="Chlorine",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_MILLIVOLT,
+        icon="mdi:pool",
+    ),
+    SensorEntityDescription(
+        key="ph",
+        name="pH",
+        icon="mdi:pool",
+    ),
+    SensorEntityDescription(
+        key="temperature",
+        name="Water Temp",
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+    ),
+    SensorEntityDescription(
+        key="date_time",
+        name="Last Measured",
+        device_class=DEVICE_CLASS_TIMESTAMP,
+    ),
+    SensorEntityDescription(
+        key="red_ox",
+        name="Red OX",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_MILLIVOLT,
+        icon="mdi:pool",
+    ),
+)
 
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Defer sensor setup to the shared sensor module."""
-    flipr_id = config_entry.data[CONF_FLIPR_ID]
     coordinator = hass.data[DOMAIN][config_entry.entry_id]
 
-    sensors_list = []
-    for sensor in SENSORS:
-        sensors_list.append(FliprSensor(coordinator, flipr_id, sensor))
-
-    async_add_entities(sensors_list, True)
+    sensors = [FliprSensor(coordinator, description) for description in SENSOR_TYPES]
+    async_add_entities(sensors, True)
 
 
 class FliprSensor(FliprEntity, SensorEntity):
     """Sensor representing FliprSensor data."""
 
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize Flipr sensor super class."""
+        super().__init__(
+            coordinator,
+            description,
+        )
+        self.entity_description = description
```

Comment:
Set this in `FliprEntity` and you can remove the init from here.

Reply:
OK, I removed it and set it in FliprEntity.